controllers/front/SmailyCustomerCron.php

```diff
@@ -111,32 +88,117 @@ private function getUserData($customer)
         $syncronize_additonal = unserialize(Configuration::get('SMAILY_SYNCRONIZE_ADDITIONAL'));
         if (!empty($syncronize_additonal)) {
             foreach ($syncronize_additonal as $sync_data) {
-                $userdata[pSQL($sync_data)] = $customer[pSQL($sync_data)];
+                if (isset($customer[pSQL($sync_data)])) {
+                    $userdata[pSQL($sync_data)] = $customer[pSQL($sync_data)];
+                }
             }
         }
         $userdata['email'] = pSQL($customer['email']);
         return $userdata;
     }
 
     /**
-     * Get unsubscribed users list from Smaily.
+     * Get unsubscribers from smaily and change subscription status to unsubscribed in store.
+     *
+     * @param int $limit Limit request size.
      *
-     * @return array $unsubscribers Unsubscribers list from Smaily.
+     * @return bool Success status.
      */
-    private function getUnsubscribers()
+    private function removeUnsubscribers($limit = 1000)
     {
-        $data = array(
-            'list' => 2,
-        );
-        // Api call to Smaily
-        $response = $this->module->callApi('contact', $data);
-        // If successful return unsubscribers.
-        if (isset($response['success'])) {
-            return $response['result'];
-        // If has errors save errors to log.
-        } else {
-            $this->module->logTofile('smaily-cron.txt', $response['error']);
-            return array();
+        $offset = 0;
+        $is_success = true;
```

Comment:
You can distill `$is_success` variable pattern down to:
```php
function removeUnsubscribers() {
    $offset = 0;

    while (true) {
        $result = $this->apiCall();
        if ($result['success'] === false) {
            return false;  // This will break the loop and return `false` as the function's return value.
        }

        if (empty($result['result'])) {
            break;  // You can use "return true" here, but I think having "break" here is better.
        }

        $query_result = Db::getInstance()->execute($query);
        if ($query_result === false) {
            return false;
        }

        $offset++;
    }

    return true;
}
```

This works because you will never have functionality in this method that needs to run after loop has finished or failed to complete (that sort of functionality will always live in `syncContacts` method), and it gives other module developers a clear signal (with `return false`) without the need to read the rest of the code in order to know that no more code is executed from this point forward. Plus there is an added bonus that there is one less variable to track.

Reply:
Cleaned up `is_success` variable. https://github.com/sendsmaily/smaily-prestashop-module/pull/10/commits/d96b432a1ae35ba93f4e0827d7c73453d56e113d